apps/avifdec.c

```diff
@@ -301,11 +301,9 @@ int main(int argc, char * argv[])
     }
 
 cleanup:
-    if (decoder) {
-        if (returnCode != 0) {
-            avifDumpDiagnostics(&decoder->diag);
-        }
-        avifDecoderDestroy(decoder);
```

Comment:
Joe: The current code does not need this null check for `decoder`. If you prefer to keep it, you can decline this pull request.